modules/percolator/src/main/java/org/elasticsearch/percolator/ExtractQueryTermsService.java

```diff
@@ -281,10 +337,42 @@ public static Query createQueryTermsQuery(IndexReader indexReader, String queryM
         return new TermsQuery(extractedTerms);
     }
 
+    static class Result {
+
+        final Set<Term> terms;
+        final boolean verified;
+        final boolean alwaysMatch;
+
+        Result(boolean verified) {
+            this.terms = Collections.emptySet();
+            this.verified = verified;
+            this.alwaysMatch = false;
+        }
+
+        Result(boolean verified, boolean alwaysMatch) {
+            this.verified = verified;
+            this.alwaysMatch = alwaysMatch;
+            this.terms = Collections.emptySet();
+        }
+
+        Result(boolean verified, Term term) {
+            this.terms = Collections.singleton(term);
+            this.verified = verified;
+            this.alwaysMatch = false;
+        }
+
+        Result(boolean verified, Set<Term> terms) {
+            this.terms = terms;
+            this.verified = verified;
+            this.alwaysMatch = false;
+        }
```

Comment:
Can we reduce the number of constructors and make them call one another. Also I think it would be nice to have some assertions like eg. it does not make sense to have verified = false and alwaysMatch = true?